error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/PreconditionsRules.java

```diff
@@ -0,0 +1,79 @@
+package tech.picnic.errorprone.refasterrules;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.errorprone.refaster.ImportPolicy.STATIC_IMPORT_ALWAYS;
+
+import com.google.common.base.Preconditions;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.UseImportPolicy;
+
+/** Refaster templates related to statements dealing with {@link Preconditions}. */
+final class PreconditionsRules {
+  private PreconditionsRules() {}
+
+  /** Prefer {@link Preconditions#checkArgument(boolean)} over more verbose alternatives. */
+  static final class CheckArgument {
+    @BeforeTemplate
+    void before(boolean condition) {
+      if (condition) {
+        throw new IllegalArgumentException();
+      }
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(boolean condition) {
+      checkArgument(!condition);
+    }
+  }
+
+  /** Prefer {@link Preconditions#checkArgument(boolean, Object)} over more verbose alternatives. */
+  static final class CheckArgumentWithMessage {
+    @BeforeTemplate
+    void before(boolean condition, String message) {
+      if (condition) {
+        throw new IllegalArgumentException(message);
+      }
+    }
```

Comment:
Opening a thread w/ @Stephan202's message:

> The format string variants require special handling and are perhaps better deferred to another ticket (though I have some ideas around that; can write some stuff about that after the meetings I have now).

I also thought of leaving many of those as follow up. :+1: 

One thought on this however. Wouldn't e.g. a `String#format()` call already partially be handled by the case I had pointed out here? As a result, we'd get
```java
checkArgument(!condition, String.format(message, obj1));
```
which I'd then expect to be re-written by _another_ `Preconditions` rule. Curious to hear the thinking.

Reply:
Indeed! The philosophy of the Refaster rules defined in this repository are that they define the smallest/most generic possible improvement, so that in unison they improve the largest amount of code. So what you propose is the way to go.

There's one caveat here: not all `String.format` calls can be replaced. What I suspect we need is a `Matcher` that identifies valid [`Strings.lenientFormat`](https://guava.dev/releases/31.1-jre/api/docs/com/google/common/base/Strings.html#lenientFormat(java.lang.String,java.lang.Object...)) format strings; then we can instruct Refaster to rewrite only eligible expressions using `@Matches(IsLenientFormatTemplateString.class)`. (Best name TBD.)

Reply:
^ It'll be more "scalable" to handle the `String.format` logic as a `BugChecker`, so let's defer that to a separate PR.